server/src/main/java/org/jboss/as/server/operations/HttpManagementAddHandler.java

```diff
@@ -184,7 +183,15 @@ protected void installServices(OperationContext context, HttpInterfaceCommonPoli
 
             RemotingHttpUpgradeService.installServices(serviceTarget, ManagementRemotingServices.HTTP_CONNECTOR, httpConnectorName,
                     ManagementRemotingServices.MANAGEMENT_ENDPOINT, commonPolicy.getConnectorOptions());
+            if (socketBindingServiceName != null) {
+                return Arrays.asList(socketBindingServiceName, UndertowHttpManagementService.SERVICE_NAME, HTTP_UPGRADE_REGISTRY.append(httpConnectorName));
+            }
+            return Arrays.asList(UndertowHttpManagementService.SERVICE_NAME, HTTP_UPGRADE_REGISTRY.append(httpConnectorName));
+        }
+        if (socketBindingServiceName != null) {
```

Comment:
This check and the one at L186 shouldn't be needed as UndertowHttpManagementService.SERVICE_NAME will depend on the socket binding service if this var isn't null.